Source/WebKit/Shared/Cocoa/CoreIPCCVPixelBufferRef.serialization.in

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if PLATFORM(COCOA)
+
+webkit_platform_headers: "CoreIPCCVPixelBufferRef.h"
+[CustomHeader, RValue, WebKitPlatform] class WebKit::CoreIPCCVPixelBufferRef {
+    mach_port_t leakSendRight();
```

Comment:
Does this actually leak anything?  Can't we just use the MachSendRight serialization instead of going all the way to mach_port_t?

Reply:
I've been unable to get it to work with the MachSendRight serialization. Regardless of if the MachSendRight is created with create or adopt in sendRightFromPixelBuffer, the test won't be able to decode the reply.

Maybe I should dig deeper to figure out how to make it work though.

Reply:
I finally figured this out. I couldn't use the MachSendRight serialization because `SerializationTestSender::performSendWithAsyncReplyWithoutUsingIPCConnection` does not pass the "attachments" from the encoder to the decoder. I'll fix that issue in this PR and use the MachSendRight serialization.